core/src/test/java/org/pentaho/di/core/vfs/KettleVFSTest.java

```diff
@@ -52,23 +48,86 @@ public void testStartsWithScheme() {
 
 
   @Test
-  public void testCheckForSchemeSuccess() throws KettleFileException, IOException {
-    String[] schemes = {"hdfs"};
+  public void testCheckForSchemeSuccess() throws KettleFileException {
     String vfsFilename = "hdfs://hsbcmaster.labs.eag.hitachivantara.com:8020/tmp/acltest/";
 
-    boolean test = KettleVFS.checkForScheme(schemes, true, vfsFilename, null, null);
-    assertFalse( test );
+    boolean test = "hdfs".equals( KettleVFS.getScheme( vfsFilename ) );
+    assertTrue( test );
+  }
 
+  @Test( expected = KettleFileException.class )
+  public void testCheckForSchemeFailWithException() throws KettleFileException {
+    String invalidVfsFilename = "{\"sport_hitting_tm\":{\"copyRight\":\" Copyright 2019 MLB Advanced Media, L.P.  Use of any"
```

Comment:
Since this JSON string is copyrighted, should we use it in a public test case in our code?  It might be better to avoid copyright issues and mock up a JSON string instead...

Reply:
Thank you Chris, i totally agree with you, it went by the test door.